### PR TITLE
Add GraphicalHost assembly to enable Out-GridView, Show-Command, and Get-Help -ShowWindow

### LINKED ISSUE #9778: Help command in PS 6.2.1 has a non-functional -ShowWindow parameter
# Steps to reproduce

```powershell
help Compress-Archive -ShowWindow
#works with any command, not just Compress-Archive
```

# Expected behavior

```powershell
help Compress-Archive -ShowWindow
#expect a new window to pop up displaying the help for the cmdlet.
#Since this is really a Windows platform thing, then it would be ok if this parameter simply was not there

```

# Actual behavior

```powershell
PS C:\Program Files\PowerShell\6> help Compress-Archive -ShowWindow
Get-Help : A parameter cannot be found that matches parameter name 'ShowWindow'.
At line:64 char:22
+     $help = Get-Help @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [Get-Help], ParameterBindingException
+ FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.PowerShell.Commands.GetHelpCommand

```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.1
PSEdition                      Core
GitCommitId                    6.2.1
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### LINKED ISSUE #10599: Out-GridView not working as expected on PowerShell 7
There is a script which retrieves some information from a database and returns values on console. When I run this script to fetch information and pipe it to Out-GridView it works just fine on PowerShell 5.1.18362.145.

I installed PowerShell 7 (7.0.0-preview.4) and installed Graphical tools (Microsoft.PowerShell.GraphicalTools)from Powershell Gallery, It is now able to return the value fine on console but when it is piped to Out-GridView, It throws exception.

# Steps to reproduce

```powershell

./Script-Name -ComputerName Server1.domail.com | Out-GridView
```

# Expected behavior

```It should redirect the output to Grid view but it fails with an exception.
Output on console with PowerShell 5.1, Works for Out-GridView too,
system.hostname : Server1.domain.com
owner Name      : PRAKASH KUMAR
owner Email     : prakash@email.com
os_family       : windows
os_version      : 2012R2
platform        : vmware
Status          : Running
Added at        : Mar 20, 2017 11:10:23
landscape       : Dev
purpose         : testing
stability_zone  : 2


Console Output on PowerShell 7, Not Working on Out-GridView

system.hostname : Server1.domain.com
owner Name      : PRAKASH KUMAR
owner Email     : prakash@email.com
os_family       : windows
os_version      : 2012R2
Platform        : vmware
Status          : Running
Added at        : Mar 20, 2017 11:10:23
landscape       : Dev
purpose         : testing
stability_zone  : 2

```

# Actual behavior

```
Exception while executing with Out-GridView

ogv : At line:1 char:10
+ $_.owner Name
+          ~~~~
Unexpected token 'Name' in expression or statement.
At line:1 char:64
+ .\ITC-Query.ps1 -ComputerName or1010050152190.corp.adobe.com  |ogv
+                                                                ~~~
+ CategoryInfo          : ParserError: (:) [Out-GridView], ParseException
+ FullyQualifiedErrorId : UnexpectedToken,OutGridView.Cmdlet.OutGridViewCmdletCommand

PS C:\Work\Working>

```

# Environment data

```
Not Working for Out-GridView
Name                           Value
----                           -----
PSVersion                      7.0.0-preview.4
PSEdition                      Core
GitCommitId                    7.0.0-preview.4
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


Working for Out-GridView
Name                           Value
----                           -----
PSVersion                      5.1.18362.145
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.145
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Add back GraphicalHost assembly code to enable Out-GridView, Show-Command, and Get-Help -ShowWindow on Windows only.

Some of the original code was changed for updated copyright and some style issues.  Because of a difference in the Windows build system vs dotnet msbuild on embedding resources, had to change resgen to work with what XAML expects.  I would focus the code review on source code that was already in this repo that had to be changed to make this work.  The original WPF code is largely unchanged except where needed to get it to compile.

No automated tests so I did some manual validation for all three commands.

In the future, we'll probably want to split this out as a separate module, but currently part of the code is in SMA.dll and the rest is in GraphicalHost.dll.  Removing the SMA.dll code is not trivial as we need to expose new public APIs for the module in how the cmdlet talks to the WPF window.  Also the cmdlet is part of Utility module which is a breaking change in itself.

Note the source folder is the namespace name as resgen uses that to produce the path to the resources and the code expects that path.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/10599
Fix https://github.com/PowerShell/PowerShell/issues/9778

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5002
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -2210,7 +2210,7 @@ function Start-CrossGen {
     $crossGenRequiredAssemblies = @("mscorlib.dll", "System.Private.CoreLib.dll")
```
Comment (PoshChan) [inline build.psm1]: @SteveL-MSFT, your last commit had 4 failures in `PowerShell-CI-windows`
Verify approved aliases list.All approved aliases present (no new aliases added, no aliases removed)
```powershell
Expected $null or empty, but got @(@{InputObject=ogv -> Out-GridView; SideIndicator=<=}, @{InputObject=shcm -> Show-Command; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 553
553:         $result | Should -BeNullOrEmpty
```
Verify approved aliases list.All approved aliases have the correct 'ReadOnly' option
```powershell
Expected $null or empty, but got @(@{InputObject=ogv -> Out-GridView; SideIndicator=<=}, @{InputObject=shcm -> Show-Command; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 577
577:         $result | Should -BeNullOrEmpty
```
Verify approved aliases list.All approved Cmdlets present (no new Cmdlets added, no Cmdlets removed)
```powershell
Expected $null or empty, but got @('Out-GridView', 'Show-Command').
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 584
584:         $result | Should -BeNullOrEmpty
```
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Write-Verbose; ConfirmImpact=None; SideIndicator=<=}, @{Name=Write-Warning; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

Comment (PoshChan) [inline build.psm1]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Out-GridView; ConfirmImpact=; SideIndicator==>}, @{Name=Show-Command; ConfirmImpact=; SideIndicator==>}, @{Name=Show-Command; ConfirmImpact=None; SideIndicator=<=}, @{Name=Tee-Object; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

Comment (PoshChan) [inline build.psm1]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Out-GridView; ConfirmImpact=; SideIndicator==>}, @{Name=Tee-Object; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

**Comment by SteveL-MSFT** [comment]
Marking it WIP as I work through the Codacy and Codefactor issues

**Comment by Jaykul** [comment]
I still think this should be an external module 😞

**Comment by SteveL-MSFT** [comment]
@Jaykul did you read the PR description above?

**Comment by SteveL-MSFT** [comment]
@iSazonov did you have a script to fix all the documentation issues previously?

File: assets/files.wxs [hunk 2]
```diff
@@ -3065,6 +3065,15 @@
       <Component Id="cmpEC42F4F98D70437BB0E290A0E0FE1775" Guid="{5483092a-12c2-4e6c-b730-be491b8909c0}">
```
Comment (PoshChan) [inline assets/files.wxs]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Breakpoint SDK Unit Tests.Handling errors while managing breakpoints in a remote runspace via the SDK.Error occurred in Context block
```powershell
Exception calling "RemoveBreakpoint" with "1" argument(s): "The background process reported an error with the following message: Remote transport error: ."
At /home/vsts/work/1/s/test/powershell/SDK/Breakpoint.Tests.ps1:154 char:13
```

**Comment by SteveL-MSFT** [comment]
@PoshChan please retry linux

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-Linux`

**Comment by iSazonov** [comment]
> @iSazonov did you have a script to fix all the documentation issues previously? also making the using directives alphabetical.

For second https://github.com/PowerShell/PowerShell/pull/9490

Fixing doc issues was done manually. I can do this in follow PR.

**Comment by SteveL-MSFT** [comment]
@iSazonov Seems like some of the doc rules can be scriptable, let me give it a shot since there's so many.  Thanks for the `using` script.

**Comment by iSazonov** [comment]
For doc issues I used many Regex queries in VS Code and Notepad++.

File: src/Microsoft.Management.UI.Internal/ShowCommand/ShowCommandSettings.Designer.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿//------------------------------------------------------------------------------
+//------------------------------------------------------------------------------
 // <auto-generated>
 //     This code was generated by a tool.
 //     Runtime Version:4.0.30319.16808
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ShowCommand/ShowCommandSettings.Designer.cs]: Should we add copyright header?

**Comment by iSazonov** [comment]
> I would focus the code review on source code that was already in this repo that had to be changed to make this work. The original WPF code is largely unchanged except where needed to get it to compile.

Maybe add original code in first PR and then add new commits in follow PRs? It is simplify code review.

**Comment by SteveL-MSFT** [comment]
@iSazonov I'm ok deferring the code cleanup.  However, it's a large amount of code.  I'd rather focus this effort on the changes to files outside of the graphicalhost code base since that code base is from Windows PowerShell and has been working for some time.  I don't want to invest in improving GraphicalHost, this is purely a parity with Windows PowerShell effort.  I'd rather invest in a cross platform version of the tools over time.

**Review by iSazonov — APPROVED** [review]
Reviewed commit by commit.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/OutGridView/OutGridViewCommand.cs [hunk 4]
```diff
@@ -312,7 +312,7 @@ internal GridHeader(OutGridViewCommand parentCmd)
             internal static GridHeader ConstructGridHeader(PSObject input, OutGridViewCommand parentCmd)
             {
                 if (DefaultScalarTypes.IsTypeInList(input.TypeNames) ||
-                    OutOfBandFormatViewManager.IsPropertyLessObject(input))
+                    !OutOfBandFormatViewManager.HasNonRemotingProperties(input))
```
Comment (sdwheeler) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/OutGridView/OutGridViewCommand.cs]: Please update FWLink on line 39. New link id is 2109378

File: assets/files.wxs (hunk 2 above)
Comment (PoshChan) [inline assets/files.wxs]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Online help tests for PowerShell Cmdlets.Validate 'get-help Out-GridView -Online'
```powershell
Expected strings to be the same, but they were different.
Expected length: 46
Actual length:   47
Strings differ at index 40.
Expected: '...nkID=11336...'
But was:  '...nkID=21093...'
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Help\HelpSystem.OnlineHelp.Tests.ps1: line 42
42:                 $actualURI | Should -Be $cmdlet.HelpURI
```

**Comment by SteveL-MSFT** [comment]
@sdwheeler does Show-Command also need a new fwlink?

File: assets/files.wxs (hunk 2 above)
Comment (PoshChan) [inline assets/files.wxs]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.http://shop.oreilly.com/product/9780596801519.do should work
```powershell
retry of URL failed with error: Response status code does not indicate success: 504 (Gateway Time-out).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

**Comment by sdwheeler** [comment]
@SteveL-MSFT Show-Command https://go.microsoft.com/fwlink/?linkid=2109589

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT please resolve merge conflict.

Also, open a PR in the PowerShell-Native repo to add the new assembly in the trusted assemblies list:

https://github.com/PowerShell/PowerShell-Native/blob/d4acbd519d1c77e07abc894db42853055ef65f55/src/powershell-native/nativemsh/pwrshcommon/pwrshcommon.cpp#L684-L686

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan does it make sense to add it since it can't be used remotely?

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT merging this PR. I can confirm with @PaulHigin if the change in trusted assemblies must be made. It does not block this PR.

**Comment by RichardSiddaway** [comment]
Get-Help -ShowWindow only shows the first paragraph of text for each parameter. This is a bug from Windows PowerShell that seems to have been carried over

**Comment by iSazonov** [comment]
@RichardSiddaway No plans exist to fix the old code. All new graphical cmdlets will be in https://github.com/PowerShell/GraphicalTools Also MSFT team has a plan to rewrite Help module.

**Comment by SteveL-MSFT** [comment]
@RichardSiddaway, @iSazonov is correct, this is simply a straight port with no plans to invest in these tools.  They are added for parity with Windows PowerShell.

**Comment by RichardSiddaway** [comment]
That's a shame that a long standing bug has been allowed to propagate into v7

**Comment by iSazonov** [comment]
@RichardSiddaway We are not happy too. I hope there are people who will port the cmdlets. Welcome to these projects.

**Comment by yecril71pl** [comment]
https://github.com/PowerShell/GraphicalTools/issues/29

**Pull request merged**